pgx/src/iterators.rs

```diff
@@ -0,0 +1,129 @@
+use pgx_utils::sql_entity_graph::metadata::{
+    ArgumentError, ReturnVariant, ReturnVariantError, SqlTranslatable, SqlVariant,
+};
+
+use crate::{pg_sys, IntoDatum};
+
+pub struct SetOfIterator<'a, T>
```

Comment:
`'a` binds the lifetime of the iterator, but the iterator is in a Box, so when is the lifetime not actually the lifetime of `T`?

Reply:
I believe this was for `'static` in certain cases.

Reply:
I wonder if GATs can make this go away, in effect.